core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -373,6 +407,49 @@ private[spark] class SparkSubmit extends Logging {
       localPyFiles = Option(args.pyFiles).map {
         downloadFileList(_, targetDir, sparkConf, hadoopConf, secMgr)
       }.orNull
+
+      if (isKubernetesClient &&
+        sparkConf.getBoolean("spark.kubernetes.submitInDriver", false)) {
```

Comment:
What's the purpose of checking `spark.kubernetes.submitInDriver`, which AFAIK is used to indicate the cluster mode.

Reply:
I need to make sure i run this only in cluster mode ad the second submit time where we are using client mode.

Reply:
Got it.